### ISSUE TITLE
[dev.icinga.com #7973] Provide a text description when the user provides information that is not in the list of allowed values

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1243]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/7973

**Created by mhein on 2014-12-03 11:52:04 +00:00**

Assignee: _jmeyer_
Status: _Resolved_  (closed on _2015-02-12 08:13:00 +00:00_)
Target Version: _2.0.0-beta3_
Last Update: _2015-02-12 08:13:00 +00:00 (in Redmine)_

---

http://www.w3.org/TR/WCAG20-TECHS/G84.html wrote:
> When users enter input that is validated, and errors are detected, the nature of the error needs to be described to the user in manner they can access. One approach is to present an alert dialog that describes fields with errors when the user attempts to submit the form. Another approach, if validation is done by the server, is to return the form (with the user's data still in the fields) and a text description at the top of the page that indicates the fact that there was a validation problem, describes the nature of the problem, and provides ways to locate the field(s) with a problem easily. The "in text" portion of the Success Criterion underscores that it is not sufficient simply to indicate that a field has an error by putting an asterisk on its label or turning the label red. A text description of the problem should be provided. \[...\]

When input must be one of a set of allowed values, the text description should indicate this fact. It should include the list of values if possible, or suggest the allowed value that is most similar to the entered value. Currently we display something like "was not found in the haystack".

**Changesets**

_2015-02-12 08:11:02 +00:00_ by _jmeyer_ 33c75a833079b026c286da69ef2c75209259229f

    Add String::findSimilar()
    
    refs #7973

_2015-02-12 08:11:02 +00:00_ by _jmeyer_ a630db93e867507aeede0066729afe7d3861a9cc

    Show proper examples in case a user doesn't enter a permitted/correct val
    
    resolves #7973

---

**Parent Task:** [#8415](https://dev.icinga.com/issues/8415)
